.github/workflows/build-test-deploy.yml

```diff
@@ -637,14 +638,78 @@ jobs:
           docker tag composable-devnet-container:latest "${{ env.DOCKER_REGISTRY_NAME }}/composable-sandbox:latest"
           docker push "${{ env.DOCKER_REGISTRY_NAME }}/composable-sandbox:latest"
 
-  devcontainer:
-    name: "Docker Devcontainer"
+  devcontainer-build:
+    name: "Devcontainer build"
+    needs:
+      - devcontainer-minimal-build
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    concurrency:
+      group: ${{ github.workflow}}-devcontainer-build-${{ github.ref }}
+      cancel-in-progress: true
+    container:
+      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
+        with:
+          skipPush: true
+          installCommand: |
+            nix-channel --add ${{ env.NIX_NIXPKGS_CHANNEL }} nixpkgs
+            nix-channel --update
+            nix-env -iA nixpkgs.cachix nixpkgs.docker nixpkgs.docker-buildx
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: ${{  env.NIX_CACHIX }}
+      - run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#packages.x86_64-linux.devcontainer --no-update-lock-file --show-trace -L
+
+  devcontainer-minimal-build:
     needs:
+      - nixfmt-check  
+    runs-on: 
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    concurrency:
+      group: ${{ github.workflow}}-devcontainer-minimal-build-${{ github.ref }}
+      cancel-in-progress: true  
+    steps:
+    - name: Clean up
+      continue-on-error: true
+      run: |
+        sudo chown -R $USER:$USER $GITHUB_WORKSPACE
+        docker system prune --force --all --volumes
+    - uses: actions/checkout@v3
+      with: 
+        fetch-depth: 0
+    - name: Login to DockerHub
+      uses: docker/login-action@v1 
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}  
+    - name: composable-devcontainer-minimal
+      run: |
+        git config --global --add safe.directory /__w/composable/composable
+        COMMIT_SHA=$(git rev-parse --short=9 HEAD)
+        docker build --file Dockerfile --tag ${{env.DOCKER_REGISTRY_NAME}}/composable-devcontainer-minimal:${COMMIT_SHA} --tag ${{env.DOCKER_REGISTRY_NAME}}/composable-devcontainer-minimal:latest .
+        docker push --all-tags ${{env.DOCKER_REGISTRY_NAME}}/composable-devcontainer-minimal
```

Comment:
Are you going to upload `${{env.DOCKER_REGISTRY_NAME}}/composable-devcontainer-minimal:latest` for every commit of a PR?

Reply:
yes for now. i need many tests and iteration before done. will gate it later.

Reply:
until it will replace composable-devcontainer i guess

Reply:
commented on that. so will keep it publish each time until we have more frozen features